app/mixins/build-favicon.coffee

```diff
@@ -0,0 +1,36 @@
+`import Ember from 'ember'`
+`import { colorForState } from 'travis/utils/helpers'`
+`import FaviconManager from 'travis/utils/favicon-manager'`
+
+Mixin = Ember.Mixin.create
+  actions:
+    faviconStateDidChange: (state) ->
+      if state
+        @setFaviconForState(state)
+      else
+        @setDefault()
+
+  init: ->
+    @faviconManager = new FaviconManager()
+
+    @_super.apply this, arguments
+
+  setFaviconForState: (state) ->
+    color = colorForState(state)
+
+    images = {
```

Comment:
You can find all the favicons here: https://github.com/travis-ci/travis-web/tree/master/public